src/get-new-changes.test.ts

```diff
@@ -0,0 +1,451 @@
+        .mockResolvedValueOnce('Update Attributions') // Already logged
+        .mockResolvedValueOnce('Bump version'); // Already logged
+
+      const result = await getNewChangeEntries({
+        mostRecentTag: 'v1.0.0',
+        repoUrl,
+        loggedPrNumbers: ['12345'],
+        loggedDescriptions: ['Update Attributions', 'Bump version'],
+        useChangelogEntry: false,
+        useShortPrLink: false,
+      });
+
+      // Should return empty array - all duplicates
+      expect(result).toHaveLength(0);
+    });
+  });
+
+  describe('edge cases', () => {
+    it('should handle no commits since last tag', async () => {
+      mockRunCommandAndSplit.mockResolvedValueOnce([]); // No commits
+
+      const result = await getNewChangeEntries({
+        mostRecentTag: 'v1.0.0',
+        repoUrl,
+        loggedPrNumbers: [],
+        loggedDescriptions: [],
+        useChangelogEntry: false,
+        useShortPrLink: false,
+      });
+
+      expect(result).toHaveLength(0);
+    });
+
+    it('should handle null as most recent tag', async () => {
+      mockRunCommandAndSplit.mockResolvedValueOnce(['commit1']);
+      mockRunCommand.mockResolvedValueOnce('feat: feature (#12345)');
+
+      const result = await getNewChangeEntries({
+        mostRecentTag: null, // No previous tag
+        repoUrl,
+        loggedPrNumbers: [],
+        loggedDescriptions: [],
+        useChangelogEntry: false,
+        useShortPrLink: false,
+      });
+
+      expect(result).toHaveLength(1);
+      expect(result[0].description).toContain('12345');
+    });
+  });
+
+  describe('formatting with PR links', () => {
+    it('should include PR link when useShortPrLink is false', async () => {
+      mockRunCommandAndSplit.mockResolvedValueOnce(['commit1']);
+      mockRunCommand.mockResolvedValueOnce('feat: add feature (#12345)');
+
+      const result = await getNewChangeEntries({
+        mostRecentTag: 'v1.0.0',
+        repoUrl,
+        loggedPrNumbers: [],
+        loggedDescriptions: [],
+        useChangelogEntry: false,
+        useShortPrLink: false,
+      });
+
+      expect(result).toHaveLength(1);
+      expect(result[0].description).toContain(
+        '[#12345](https://github.com/MetaMask/metamask-mobile/pull/12345)',
+      );
+    });
+
+    it('should use short PR link when useShortPrLink is true', async () => {
+      mockRunCommandAndSplit.mockResolvedValueOnce(['commit1']);
+      mockRunCommand.mockResolvedValueOnce('feat: add feature (#12345)');
+
+      const result = await getNewChangeEntries({
+        mostRecentTag: 'v1.0.0',
+        repoUrl,
+        loggedPrNumbers: [],
+        loggedDescriptions: [],
+        useChangelogEntry: false,
+        useShortPrLink: true,
+      });
+
+      expect(result).toHaveLength(1);
+      expect(result[0].description).toContain('(#12345)');
+      expect(result[0].description).not.toContain('[#12345]'); // Should not have brackets
```

Comment:
Again it would be easier to understand exactly what `getNewChangeEntries` is giving us if we used the full return value here:

``` suggestion
      expect(result).toStrictEqual([
        {
          description: 'feat: add feature (#12345)',
          subject: 'feat: add feature (#12345)',
        },
      ]);
```

It's also easy to see that this isn't really a link at all.)